test/LaunchDarkly.EventSource.Tests/EventSourceEndToEndTest.cs

```diff
@@ -29,100 +29,78 @@ public void ReceiveEventStreamInChunks()
             // parsed correctly regardless of how the chunks line up with the events.
 
             var eventData = new List<string>();
-            var chunks = new List<string>();
             for (var i = 0; i < 200; i++)
             {
                 eventData.Add(string.Format("data{0}", i) + new string('x', i % 7));
             }
             var allBody = string.Concat(eventData.Select(data => "data:" + data + "\n\n"));
-            for (var pos = 0; ;)
+            var allEventsReceived = new EventWaitHandle(false, EventResetMode.ManualReset);
```

Comment:
Here I added some synchronization (see line 54 and line 80) to ensure that the HTTP handler won't exit before the client side of the test has caught up to it.